src/transformers/trainer.py

```diff
@@ -702,6 +710,8 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
         self.callback_handler.optimizer = self.optimizer
         self.callback_handler.lr_scheduler = self.lr_scheduler
         self.callback_handler.train_dataloader = train_dataloader
+        trial_name = self.hp_name(trial) if self.hp_name is not None else None
+        self.callback_handler.trial_info = {"trial": self._trial, "trial_name": trial_name}
```

Comment:
Should this be in the state instead?

Reply:
Yes, did not want to change the state, to limit the amount of changes, but I can move it to the state.

Reply:
Resolved in https://github.com/huggingface/transformers/pull/7935/commits/3179a1f8044446fd89275a9fbde82a2fa4447ce1